File: backend/src/main/java/org/woork/backend/posting/Categories.java
package org.woork.backend.posting;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Stream;

public enum Categories {
    JARDINERIA("8DV7AS"),
    PLOMERIA("4B2MK6"),
    ELECTRICIDAD("1P0HBC"),
    MASCOTAS("2HZ3PL"),
    HOGAR("G6S7JK"),
    VEHICULOS("0PJD4M"),
    CARPINTERIA("M6BZK0"),
    ALBANILERIA("R2ZLKA"),
    ALBERCAS("8NASND"),
    CANCELERIA("KM0Q29"),
    DECORACIONES("P10CSA"),
    PINTURA("NBJD0S"),
    IMPERMEABLIZADO("3NMAZS"),
    MUEBLES("JK89XS"),
    INSTALACION("BHDUS3");

    public final String code;

    Categories(String code) {
        this.code = code;
    }

    @Override
    public String toString() {
        return code;
    }

    public static String[] getNames() {
        return Stream.of(Categories.values())
                .map(Categories::name)
                .toArray(String[]::new);
    }

    public static List<String> getCodes() {
        return Stream.of(Categories.values()).map(Categories::toString).toList();
    }

    private static final Map<String, Categories> stringToCategory = new HashMap<>();
    static {
        for(Categories type : Categories.values()) {
            stringToCategory.put(type.toString(), type);
        }
    }

    public static Categories fromCode(String code) {
        return stringToCategory.get(code);
    }
}

File: frontend/src/app/Services/Categories.ts
import BalloonSVG from "@/components/SVGs/JobCategories/Balloon";
import BricksSVG from "@/components/SVGs/JobCategories/Bricks";
import CouchSVG from "@/components/SVGs/JobCategories/Couch";
import DoorSVG from "@/components/SVGs/JobCategories/Door";
import DropSVG from "@/components/SVGs/JobCategories/Drop";
import Electrical from "@/components/SVGs/JobCategories/Electrical";
import Hammer from "@/components/SVGs/JobCategories/Hammer";
import House from "@/components/SVGs/JobCategories/House";
import PaintSVG from "@/components/SVGs/JobCategories/Paint";
import Pet from "@/components/SVGs/JobCategories/Pet";
import Plant from "@/components/SVGs/JobCategories/Plant";
import Plumbery from "@/components/SVGs/JobCategories/Plumbery";
import Screwdriver from "@/components/SVGs/JobCategories/Screwdriver";
import Vehicle from "@/components/SVGs/JobCategories/Vehicle";
import WaterSVG from "@/components/SVGs/JobCategories/Water";
import React from "react";

export const Categories = {
    "Jardinería": "8DV7AS",
    "Plomería": "4B2MK6",
    "Electricidad": "1P0HBC",
    "Mascotas": "2HZ3PL",
    "Hogar": "G6S7JK",
    "Vehículos": "0PJD4M",
    "Carpintería": "M6BZK0",
    "Albañilería": "R2ZLKA",
    "Albercas": "8NASND",
    "Cancelería": "KM0Q29",
    "Decoraciones": "P10CSA",
    "Pintura": "NBJD0S",
    "Impermeablizado": "3NMAZS",
    "Muebles": "JK89XS",
    "Instalación": "BHDUS3"
}

type Category = {
    name: string;
    tag: string;
    icon: React.MemoExoticComponent<({ width }: { width: number; }) => React.ReactNode>;
}

export const CategoriesList: Array<Category> = [
    {
        'name': 'Jardinería',
        'tag': '8DV7AS',
        'icon': Plant
    },
    {
        'name': 'Plomería',
        'tag': '4B2MK6',
        'icon': Plumbery
    },
    {
        'name': 'Electricidad',
        'tag': '1P0HBC',
        'icon': Electrical
    },
    {
        'name': 'Mascotas',
        'tag': '2HZ3PL',
        'icon': Pet
    },
    {
        'name': 'Hogar',
        'tag': 'G6S7JK' ,
        'icon': House
    },
    {
        'name': 'Vehículos',
        'tag': '0PJD4M',
        'icon': Vehicle
    },
    {
        'name': 'Carpintería',
        'tag': 'M6BZK0',
        'icon': Hammer
    },
    {
        'name': 'Albañilería',
        'tag': 'R2ZLKA',
        'icon': BricksSVG
    }, 
    {
        'name': 'Albercas',
        'tag': '8NASND',
        'icon': WaterSVG
    },
    {
        'name': 'Cancelería',
        'tag': 'KM0Q29',
        'icon': DoorSVG
    },
    {
        'name': 'Decoraciones',
        'tag': 'P10CSA',
        'icon': BalloonSVG
    },
    {
        'name': 'Pintura',
        'tag': 'NBJD0S',
        'icon': PaintSVG
    },
    {
        'name': 'Impermeablizado',
        'tag': '3NMAZS',
        'icon': DropSVG
    },
    {
        'name': 'Muebles',
        'tag': 'JK89XS',
        'icon': CouchSVG
    },
    {
        'name': 'Instalación',
        'tag': 'BHDUS3',
        'icon': Screwdriver
    }
]